### ISSUE TITLE
regularization parameter being ignored in tf.keras dense layer initialization

### ISSUE THREAD

**Issue by mikeymezher** [tensorflow/tensorflow#33219]
An example of this issue can easily be seen on the TF 2.0 colab advanced quickstart tutorial: https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/quickstart/advanced.ipynb#scrollTo=i-2pkctU_Ci7

Changing the code block from

```
`class MyModel(Model):
  def __init__(self):
    super(MyModel, self).__init__()
    self.conv1 = Conv2D(32, 3, activation='relu')
    self.flatten = Flatten()
    self.d1 = Dense(128, activation='relu')
    self.d2 = Dense(10, activation='softmax')

  def call(self, x):
    x = self.conv1(x)
    x = self.flatten(x)
    x = self.d1(x)
    return self.d2(x)
model = MyModel()`
```

To 

```
`class MyModel(Model):
  def __init__(self):
    super(MyModel, self).__init__()
    self.conv1 = Conv2D(32, 3, activation='relu',activity_regularizer=tf.keras.regularizers.l2(100000.))
    self.flatten = Flatten()
    self.d1 = Dense(128, activation='relu',activity_regularizer=tf.keras.regularizers.l2(100000.))
    self.d2 = Dense(10, activation='softmax',activity_regularizer=tf.keras.regularizers.l2(100000.))
  def call(self, x):
    x = self.conv1(x)
    x = self.flatten(x)
    x = self.d1(x)
    return self.d2(x)
model = MyModel()`
```
    
has no effect, despite the regularization parameter being inordinately large (we should expect learning to completely stop in this case, as the penalization should force outputs to 0). 
This is also true for kernel_regularizer.
Manually adding tf.keras.layers.ActivityRegularization also has no effect.

**Comment by oanush** [comment]
As explained no error is faced after code is replaced, Thanks!

**Comment by georgesterpu** [comment]
The code in that file does not collect the regularisation losses.
I think that you need to write:
`loss += tf.add_n(model.losses)`
in the `train_step` function.

**Comment by jvishnuvardhan** [comment]
@mikeymezher Let us know whether @georgesterpu suggestion helped you in resolving the issue? Thanks!

**Comment by mikeymezher** [comment]
Thanks @georgesterpu, that recommendation worked. In retrospect, I understand why this step is needed, since regularization wouldn't effect the predict method of the model so loss due to regularization has to be added separately; but I couldn't find any documentation for this. 

Might be worth mentioning/providing an example of this somewhere in the tf.keras.regularizers api-docs.
